pyoffers/models/offer_file.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+from .core import Model, ModelManager
+
+
+class OfferFile(Model):
+    generic_methods = ('update', 'delete')
+
+
+class OfferFileManager(ModelManager):
+    model = OfferFile
+    name = 'offer_files'
+    generic_methods = (
+        'find_by_id',
+        'find_all',
+        'find_all_ids',
+        'delete',
+        'update',
+    )
+
+    def create(self, **kwargs):
+        assert 'filename' in kwargs, \
+            'You must specify ``filename`` argument, which should be a name of a readable file in current directory.'
+        file_name_w_path = kwargs.get('filename')
```

Comment:
There are 3 names for 2 things. `filename`, `file_name_w_path` and `file_name`.
let's do it like this:

```python
def create(self, path, **kwargs):
    with open(path, 'rb') as fd:
        if 'filename' not in kwargs:
            kwargs['filename'] = os.path.basename(path)
        return self._call('create', data=kwargs, files={kwargs['filename']: fd})
```
In this case it would be possible for users to specify filename. Note, that ``setdefault`` is not used, because if there is a filename then ``os.path.basename`` will be called without a reason.

Reply:
This is similar to what it was before, in this case we have two arguments for the same thing: `path` and `filename`. I would rather stick to one argument. Auxiliary variables are just required in order not to call `kwargs.get('filename')` twice.